csrc/dynamic_transform.h

```diff
@@ -213,6 +234,11 @@ class DynamicTransformConcretizationInfo {
   //! with its concretized IterType
   std::vector<std::pair<size_t, IterType>> resize_itertypes_;
 
+  //! Holds the index of the expanded TensorView in the vector returned by
+  //! initial_info_->getDynamicExpandedTensorViews(), and a corresponding vector
+  //! of bools indicating whether each axis is in fact expanded.
+  std::vector<std::pair<size_t, std::vector<bool>>> expand_axes_;
```

Comment:
nitpicking on the mapping here using index... but I see other structure in concretization info uses similar mapping. :shrug:

Reply:
Yes, using the raw pointer seems at first like it would be more convenient right; that is how this was originally implemented. However, it gets complicated when we need to clone the dynamic Fusion for concretization: at that point we need the mapping from the original pointers in the dynamic definition to the new cloned versions. We also need to be able to compare concretization info objects and hash them, since they act as cache keys. The solution we use is to attach `DynamicTransformInitialInfo` to the fusion in the form of managed data, so that it is automatically cloned when the Fusion is cloned. Then we refer to objects held by it inside the concretization info, which lets us hold those outside the Fusion, and still apply them to any Fusion clone we want.